src/modules/mc_hover_thrust_estimator/MulticopterHoverThrustEstimator.cpp

```diff
@@ -157,7 +157,22 @@ void MulticopterHoverThrustEstimator::Run()
 					ZeroOrderHoverThrustEkf::status status;
 					_hover_thrust_ekf.fuseAccZ(-local_pos.az, -local_pos_sp.thrust[2], status);
 
-					publishStatus(status);
+					hover_thrust_estimate_s status_msg;
+
+					status_msg.hover_thrust = status.hover_thrust;
+					status_msg.hover_thrust_var = status.hover_thrust_var;
+
+					status_msg.accel_innov = status.innov;
+					status_msg.accel_innov_var = status.innov_var;
+					status_msg.accel_innov_test_ratio = status.innov_test_ratio;
+					status_msg.accel_noise_var = status.accel_noise_var;
+
+					status_msg.valid = (status.hover_thrust_var < 0.001f) && (status.innov_test_ratio < 1.f);
+
+					status_msg.timestamp = hrt_absolute_time();
```

Comment:
wouldn't `local_pos.timestamp` be more correct?

Reply:
No, this is uORB metadata that will hopefully be pushed into uORB itself at some point. When looking at logged data we want the timestamp to be as close as possible to what the rest of the system would actually see (atomic with the publication). Passing timestamps through is just hiding latency and execution time we'd be able to see for "free". I'd like to change this in ekf2 and if desired we can separately log a `timestamp_sample`.